packages/core/core-flows/src/fulfillment/steps/validate-fulfillment-providers.ts

```diff
@@ -0,0 +1,117 @@
+import { ServiceZoneDTO, ShippingOptionDTO } from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  MedusaError,
+  ModuleRegistrationName,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+import { createStep, StepResponse } from "@medusajs/workflows-sdk"
+
+type FulfillmentProviderValidationInput = {
+  id?: string
+  service_zone_id?: string
+  provider_id?: string
+}
+
+export const validateFulfillmentProvidersStepId =
+  "validate-fulfillment-providers-step"
+export const validateFulfillmentProvidersStep = createStep(
+  validateFulfillmentProvidersStepId,
+  async (input: FulfillmentProviderValidationInput[], { container }) => {
+    const dataToValidate: {
+      service_zone_id: string
+      provider_id: string
+    }[] = []
+    const shippingOptionIds = input.map((d) => d.id).filter(Boolean) as string[]
+    const fulfillmentService = container.resolve(
```

Comment:
Super NIT: maybe move `shippingOptionsIds` below `fulfillmentService`, it's a bit easier to read 😄. Maybe even passing the map to `id: input.map...` would be fine as well, WDYT?